src/Tools/ExternalAccess/Razor/Features/RazorAnalyzerAssemblyResolver.cs

```diff
@@ -63,7 +63,13 @@ internal sealed class RazorAnalyzerAssemblyResolver() : IAnalyzerAssemblyResolve
             // runs in a .net based host, it's safe to always choose the .net targeted ServiceHubCore versions.
             if (!Path.GetFileName(directory.AsSpan().TrimEnd(Path.DirectorySeparatorChar)).Equals(ServiceHubCoreFolderName, StringComparison.OrdinalIgnoreCase))
             {
-                directory = Path.Combine(directory, ServiceHubCoreFolderName);
+                var serviceHubCoreDirectory = Path.Combine(directory, ServiceHubCoreFolderName);
+
+                // The logic above only applies to VS. In VS Code there is no service hub, so appending the folder would be silly.
+                if (Directory.Exists(serviceHubCoreDirectory))
+                {
+                    directory = serviceHubCoreDirectory;
+                }
```

Comment:
@chsienki this is a bit of a speculative change on my part, because it either doesn't matter whether its there, or I just always won the race, but it makes sense to me. In VS Code there is no `ServiceHubCore` folder. Does that make sense?

Also, doesn't matter if this isn't perfect for this PR, there will be more to come :)

Reply:
Yep, I'm fine with this. A secondary question to which I suspect I know the answer: do we have integration tests in Razor for VSCode? I would have expected this to break them if we do.

Reply:
We do, in the VS Code repo, though none with cohosting turned on (yet?).

I'm thinking of reverting this change though, as it was a bit speculative when I was having all sorts of source generator loading issues, so I was doing a bit of whack-a-mole to try to fix them.

Reply:
If there is no ServiceHubCore folder in VS, then I think this is fixing a legitimate issue. If we're redirect to a non-existent assembly, the later logic will fail, and we'll ultimately fall back to loading from the SDK, meaning our redirection is broken.